src/py/mat3ra/made/tools/build/defect/terrace/analyzer.py

```diff
@@ -0,0 +1,108 @@
+from typing import List
+
+import numpy as np
+
+from mat3ra.made.tools.analyze.slab import SlabMaterialAnalyzer
+from mat3ra.made.tools.build import MaterialWithBuildMetadata
+from mat3ra.made.utils import get_atomic_coordinates_extremum
+
+
+class TerraceMaterialAnalyzer(SlabMaterialAnalyzer):
```

Comment:
We need to add documentation here about the process of the terrace creation and the parameters that this class extracts - there seem to be many